src/bit_sync.c

```diff
@@ -41,8 +42,8 @@ void bit_sync_init(bit_sync_t *b, gnss_signal_t sid)
   memset(b, 0, sizeof(bit_sync_t));
   b->bit_phase_ref = BITSYNC_UNSYNCED;
 
-  assert(sid.band == BAND_L1);
-  switch (sid.constellation) {
+  assert(sid.code == CODE_GPS_L1CA || sid.code == CODE_SBAS_L1CA);
```

Comment:
For now, yes, those are the signal types with defined bit lengths (see top of file). Eventually we should store the bit lengths for all codes in some central location and remove the asserts.